### ISSUE TITLE
Incorrect filename after uploading to (Linode) S3 disk

### ISSUE THREAD

**Issue by kurucu** [laravel-filament/filament#2348]
### Package

filament/forms

### Package Version

v2.11.12

### Laravel Version

v9.11.0

### Livewire Version

v2.10.5

### PHP Version

PHP 8.1.5

### Bug description

Using the FileUpload component, with an S3 disk (configured for Linode Object Storage), the path stored in the database by Forms is different to that for the uploaded object found at Linode.

I have tried various things e.g. setting visibility to public (but noting this is default, removed it below).

Jetstream's profile photo disk is the same disk, and works fine which rules out a number of issues. This appears to be a FileUpload bug.

```
FileUpload::make('photo_path')
                        ->image()
                        ->directory('photos'),
```

### Steps to reproduce

1. Create a form with a FormUpload component.
2. Set the (default) disk to be an S3 (might be a Linode S3 specific) disk
3. Upload file and save.
4. File cannot be found at the location saved by Forms.

Disk config options are as follows (in my case verified by Jetstream working as expected):

```
'linode' => [
            'driver' => 's3',
            'key' => env('LINODE_KEY'),
            'secret' => env('LINODE_SECRET'),
            'endpoint' => env('LINODE_ENDPOINT'),
            'region' => env('LINODE_REGION'),
            'bucket' => env('LINODE_BUCKET'),
        ],

```

### Relevant log output

```shell
Browser finds a 403 because the object doesn't exist and listing is not allowed.
Inspection of the bucket sees a new object with the right content, but a different name to that provided and saved by Forms.
```

**Comment by ryangjchandler** [comment]
@kurucu `->preserveFilenames()`?

**Comment by kurucu** [comment]
I'm not sure that's it, I don't want to keep the users file name.

The upload seems to generate one random filename, and then somewhere along the route another one is generated and stored to the database?

**Comment by danharrin** [comment]
I don't think it's Filament's responsibility to ensure compatibility with every single possible S3-"compatible" filesystem, and this is very likely to actually be a Livewire issue rather than Filament.

We don't have the resources to purchase storage on a platform for testing purposes, like Linode. If you'd like to see this fixed, you're gonna need to do some debugging yourself. Visit the `BaseFileUpload` class and check out the code that stores the file. Is the filename correct in `TemporaryFileUpload`? Is it returned correctly from the `store()` method?

**Comment by kurucu** [comment]
ok @danharrin, I'll see what I can do!

**Comment by kurucu** [comment]
OK

It has nothing to do with Linode, S3 or the disk type at all. Moving to from local to Linode, I invalidated (staging) file paths in the DB. I didn't think this would be an issue, but it was the root of the problem.

There's now an invalid path in the db. I visit the edit form, where Filament Forms is, and it just says "upload a file" (rather than showing one). Seems fine, so I click a new file hit upload and all seems good until I have my mismatched filenames.

Except I now see that all I'm getting is the same, invalid filename over and over. It's not generating two and storing a random one, it's just the same old one.

Looking at the network inspector, Livewire thinks there's a valid file. Even though file pond shows nothing, and allows me to upload a new file as if from scratch, it appears to be appending that to the old, invalid file. So I now have an array of files, except it's a single-file uploader. So the first (old one) gets stored and not the new one. **I think**.

Setting the path to null in the database, everything works as it should, no matter what filesystem I use.

Now it is an edge case. But it appears to be a Filament issue - in combination with Filepond - that when a "new" file is uploaded for a single file upload, on top of an incorrect old file, the old one gets persisted instead of the new one because it just takes the first in the array (filepond didn't know it was invalid, found nothing so showed an upload screen, and then Filament/Livewire saw a new file get uploaded and just appended it rather than replacing it).

Is that enough to get you rolling? I could try to look further, but I will be much slower and worse than you guys!

**Comment by danharrin** [comment]
So it's a duplicate of #2419 ?

**Comment by kurucu** [comment]
🤦‍♂️ Yep, it's a duplicate of #2419

**Comment by danharrin** [comment]
No worries :)

**Comment by danharrin** [comment]
Fixed by #2487.
